tools/analysis_tools/eval_metric.py

```diff
@@ -35,53 +24,27 @@ def parse_args():
         'It also allows nested list/tuple values, e.g. key="[(a,b),(c,d)]" '
         'Note that the quotation marks are necessary and that no white space '
         'is allowed.')
-    parser.add_argument(
-        '--eval-options',
-        nargs='+',
-        action=DictAction,
-        help='custom options for evaluation, the key-value pair in xxx=yyy '
-        'format will be kwargs for dataset.evaluate() function')
     args = parser.parse_args()
     return args
 
 
 def main():
     args = parse_args()
 
-    cfg = Config.fromfile(args.config)
-
-    # replace the ${key} with the value of cfg.key
-    cfg = replace_cfg_vals(cfg)
+    register_all_modules(init_default_scope=True)
 
-    # update data root according to MMDET_DATASETS
-    update_data_root(cfg)
-
-    assert args.eval or args.format_only, (
-        'Please specify at least one operation (eval/format the results) with '
-        'the argument "--eval", "--format-only"')
-    if args.eval and args.format_only:
-        raise ValueError('--eval and --format_only cannot be both specified')
+    cfg = Config.fromfile(args.config)
 
     if args.cfg_options is not None:
         cfg.merge_from_dict(args.cfg_options)
-    cfg.data.test.test_mode = True
 
-    dataset = build_dataset(cfg.data.test)
-    outputs = load(args.pkl_results)
+    dataset = DATASETS.build(cfg.test_dataloader.dataset)
+    predictions = mmengine.load(args.pkl_results)
 
-    kwargs = {} if args.eval_options is None else args.eval_options
-    if args.format_only:
-        dataset.format_results(outputs, **kwargs)
-    if args.eval:
-        eval_kwargs = cfg.get('evaluation', {}).copy()
-        # hard-code way to remove EvalHook args
-        for key in [
-                'interval', 'tmpdir', 'start', 'gpu_collect', 'save_best',
-                'rule'
-        ]:
-            eval_kwargs.pop(key, None)
-        eval_kwargs.update(dict(metric=args.eval, **kwargs))
-        print(dataset.evaluate(outputs, **eval_kwargs))
+    evaluator = Evaluator(cfg.test_evaluator)
```

Comment:
The test set generally has no labels, and the offline evaluation test set is not reasonable, it should be val